code/main.py

```diff
@@ -8,26 +8,26 @@
 from adal.adal_error import AdalError
 from msrest.exceptions import AuthenticationError
 from json import JSONDecodeError
-from utils import AMLConfigurationException, required_parameters_provided, create_aml_cluster, create_aks_cluster, mask_parameter
+from utils import AMLConfigurationException, create_aml_cluster, create_aks_cluster, mask_parameter, validate_json, required_parameters_provided
```

Comment:
I guess we still need `required_parameters_provided` because the user either must provide `name` or `compute_type`, but both aren't required.

Reply:
Yes, only if the action creates a compute target, we need the compute_type. Otherwise not.